docs/docs_feedback_sphinxext.py

```diff
@@ -0,0 +1,97 @@
+"""A sphinx extension for collecting per doc feedback."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Dict, List, Union
+
+    from sphinx.application import Sphinx
+
+
+def _modify_rst_document_source_on_read(
+        app: Sphinx,
+        docname: str,
+        source: List[str],
+) -> None:
+    """Add info block to top and bottom of each document source.
+
+    This function modifies RST source in-place by adding an admonition
+    block at the top and the bottom of each document right after it's
+    been read from disk preserving :orphan: at top, if present.
+    """
+    admonition_type = 'important'
+    valid_admonitions = {
+        'attention', 'caution', 'danger', 'error', 'hint',
+        'important', 'note', 'tip', 'warning', 'admonition',
+    }
+
+    if admonition_type not in valid_admonitions:
+        raise ValueError(
+            'Expected admonition_type to be one of '
+            f'{valid_admonitions} but got {admonition_type}.'
+        )
+
+    questions_list = """
+      1. What problem were you trying to solve when you came to this page?
+
+      2. What content was useful?
+
+      3. What content was not useful?
```

Comment:
```suggestion
    1. What problem were you trying to solve when you came to this page?
    2. What content was useful?
    3. What content was not useful?
```

The rendered list seems to be over-indented, probably because triple-quotes preserve whitespace.  I can't view the reST output (after this extension is applied) due to the theme, but I think generally it'd be safer to declare all these literals as module level constants to avoid accidentally messing up indentation.

Reply:
I hear you but because it's also augmented with another line with context-specific var, I'm not sure if moving it on the module level is a good idea. After all, we need to keep some sort of common indentation level here and 3 lines below.

I guess I'll just experiment with dedenting both.

Reply:
I'd say: check if it's creating a blockquote in the final HTML. If it is, then you know that things are indented too far.

Reply:
It is, and I was also concern about other texts provided as indented triple quoted literals, e.g. **Did this article help?** (relative to `.. important`).  Sphinx is pretty relaxed and still renders the content even if it doesn't align with `important`, but I *think* it'd be easier to notice whitespace issues if the reST is provided as something close to unindented literals, which we can then str.format on.